models/applicantModel.js

```diff
@@ -1,28 +1,461 @@
+        additionalRoutes
+      );
+
+      // =======================================
+      // Step 1: Update Request table
+      // =======================================
+      const request_days = getRequestDays(allRoutes);
+
+      // Log old data
+      const [oldData] = await conn.query(
+        `SELECT * FROM request WHERE request_id = ?`,
+        [requestId]
+      );
+      console.log("Old data:", oldData);
+
+      const updateRequestTable = `
+        UPDATE request SET
+          notes = ?,
+          requested_fee = ?,
+          imposed_fee = ?,
+          request_days = ?,
+          last_mod_date = CURRENT_TIMESTAMP
+        WHERE request_id = ?
+      `;
+
+      await conn.execute(updateRequestTable, [
+        notes,
+        requested_fee, // Allow null values for requested_fee
+        imposed_fee, // Allow null values for imposed_fee
+        request_days, // Allow null values for request_days
+        requestId, // Use the provided requestId to update the correct record
+      ]);
+
+      // Log new data
+      const [newData] = await conn.query(
+        `SELECT * FROM request WHERE request_id = ?`,
+        [requestId]
+      );
+      console.log("New data:", newData);
+
+      // =======================================
+      // Step 2: Delete old routes
+      // =======================================
+
+      const oldRoutesIds = await conn.query(
+        `SELECT route_id FROM route_request WHERE request_id = ?`,
+        [requestId]
+      );
+
+      // Delete old route request table data related to the request
+      const deleteRouteRequest = `
+        DELETE FROM route_request WHERE request_id = ?
+      `;
+      await conn.execute(deleteRouteRequest, [requestId]);
+
+      // Delete old routes from Route_Request table
+      for (const route_id of oldRoutesIds) {
+        const deleteRoute = `
+          DELETE FROM route WHERE route_id = ?
+        `;
+        await conn.execute(deleteRoute, [route_id.route_id]);
+      }
+
+
+      // =======================================
+      // Step 3: Edit Route & Route_Request table
+      // =======================================
+
+      for (const route of allRoutes) {
+        try {
+
+          console.log("Processing route:", route);
+
+          let
+            id_origin_country,
+            id_destination_country,
+            id_origin_city,
+            id_destination_city;
+
+          // Search if the country exists in the database
+          id_origin_country = await getCountryId(conn, route.origin_country_name);
+          id_destination_country = await getCountryId(conn, route.destination_country_name);
+
+          // Search if the city exists in the database
+          id_origin_city = await getCityId(conn, route.origin_city_name);
+          id_destination_city = await getCityId(conn, route.destination_city_name);
+
+
+          // Insert into Route table
+
+          const insertRouteTable = `
+          INSERT INTO route (
```

Comment:
```suggestion
          INSERT INTO Route (
```
Another incorrect name...